source/common/upstream/upstream_impl.cc

```diff
@@ -735,7 +735,7 @@ ClusterInfoImpl::ClusterInfoImpl(
                             "envoy.extensions.upstreams.http.v3.HttpProtocolOptions"),
                         factory_context.messageValidationVisitor())),
       max_requests_per_connection_(
-          PROTOBUF_GET_WRAPPED_OR_DEFAULT(config, max_requests_per_connection, 0)),
+          http_protocol_options_->common_http_protocol_options_.max_requests_per_connection()),
```

Comment:
Need to still honor the old config setting until it is removed. If both old and new are set, throw an exception. If either is set, use the value. Add these behaviors. Wrap tests for the old field in `DEPRECATED_FEATURE_TEST` (search for examples in tests).

Reply:
Added logic for intended behaviors and deprecated test.